### PR TITLE
Create fragments for shared Quickstart content

### PR DESCRIPTION
**Pull request by kevinold**
#### Description of changes:

Abstract co-located fragments (per @reesscot) for Gen2 Quickstart content.

#### Related GitHub issue #, if available:

https://app.asana.com/0/1205212189033571/1206274691787464/f

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx [hunk 1]
```diff
@@ -13,292 +13,17 @@ export function getStaticProps(context) {
 
 This Quickstart guide will walk you through how to build a task list application with TypeScript, Next.js **App Router**, and React. If you are new to these technologies, we recommend you go through the official [React](https://react.dev/learn/tutorial-tic-tac-toe), [Next.js](https://nextjs.org/docs/getting-started/installation), and [TypeScript](https://www.typescriptlang.org/docs/handbook/typescript-from-scratch.html) tutorials first.
 
-## Prerequisites
+import prerequisites from 'src/fragments/gen2/quickstart/prerequisites.mdx';
+import createProject from 'src/fragments/gen2/quickstart/create-nextjs-project.mdx';
```
Comment (heatheramz) [inline src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx:17]: @kevinold I wonder if we should co-locate the src address with where the fragment is used. We use this convention on other docs so authors can quickly find where the fragment is filed without having to scroll back up. @reesscot is there a preference from the docs team on this format?

Comment (kevinold) [reply src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx:17]: @heatheramz No problem, I'll updated if that is the preference.

Comment (kevinold) [reply src/pages/gen2/start/quickstart/nextjs-app-router/index.mdx]: Done.

**Review by heatheramz — APPROVED** [review]
LGTM

**Review by reesscot — APPROVED** [review]
LGTM

**Pull request merged**